tests/test_check_types.py

```diff
@@ -164,6 +186,69 @@ def test_preferred_s_smoke(
     assert list(check.check(noise)) == []
 
 
+@given(PREFERRED_ITEMS_CASED_STRATEGY, PADDING_STRATEGY, st.text())
+def test_preferred_s_case_sensitive(
+    items: dict[str, str], padding: Padding, path: str
+) -> None:
+    """Return correct matches when ignore_case=False with case variations."""
+    check = Check(
+        check_type=types.PreferredFormsSimple(items=items, padding=padding),
+        path=path,
+        message="{} || {}",
+        ignore_case=False,
+    )
+
+    text = " ".join(items.keys())
+    check_type = types.PreferredFormsSimple(items=items, padding=padding)
+    results = list(check_type.check(text, check))
+
+    assert all(result.replacements in items.values() for result in results)
+    for i, result in enumerate(results):
+        key = list(items.keys())[i]
+        expected_replacement = items[key]
+
+        assert result.check_path == path
+        assert result.message == f"{expected_replacement} || {key}"
+        assert result.replacements == expected_replacement
+
+
+@given(PREFERRED_ITEMS_CASED_STRATEGY, PADDING_STRATEGY, st.text())
+def test_preferred_s_case_insensitive(
+    items: dict[str, str], padding: Padding, path: str
+) -> None:
+    """Return correct matches when ignore_case=True with case variations."""
+    normalised = {k.lower(): v for k, v in items.items()}
+    check = Check(
+        check_type=types.PreferredFormsSimple(items=items, padding=padding),
+        path=path,
+        message="{} || {}",
+        ignore_case=True,
+    )
+
+    text = "  ".join(
+        k.swapcase() if i % 2 == 0 else k.upper()
+        for i, k in enumerate(items.keys())
+    )
+
+    check_type = types.PreferredFormsSimple(items=normalised, padding=padding)
+    results = list(check_type.check(text, check))
```

Comment:
If the only results evaluated are with the `normalised` dictionary, construct the `check` with this in the first place, and use `check.check`.